src/infrastructure/collection/executor/scrape_executor.py

```
@@ -76,8 +80,95 @@ def run(
+                    self._fetch_worker_loop,
+                    worker_id=i,
+                    host_queue_map=host_queue_map,
+                    on_result=on_result,
+                    pending_discovers=pending_discovers,
+                    pending_lock=pending_lock,
+                ))
+
+            for future in as_completed(futures):
+                try:
+                    total_fetched += future.result()
+                except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)